src/components/shared-settings/createSettings.tsx

```diff
@@ -8,28 +8,30 @@ interface SettingsTexts {
     primary: string
     secondary?: string
 }
-const texts = new WeakMap<ValueRef<unknown>, SettingsTexts>()
-export function createNewSettings<T extends browser.storage.StorageValue>(
+const texts = new WeakMap<ValueRef<any>, SettingsTexts>()
+
+function createInternalSettings<T extends browser.storage.StorageValue>(
+    storage: string,
     key: string,
     initialValue: T,
-    UITexts: SettingsTexts,
+    comparer: (a: T, b: T) => boolean = (a, b) => a === b,
 ) {
-    const settings = new ValueRef(initialValue)
-    texts.set(settings, UITexts)
-
-    update()
+    const settings = new ValueRef(initialValue, comparer)
+    const instanceKey = `${storage}+${key}`
+    update(instanceKey)
     settings.addListener(async newVal => {
-        const stored = ((await browser.storage.local.get()).settings as object) || {}
+        const stored = ((await browser.storage.local.get())[storage] as object) || {}
         browser.storage.local.set({
```

Comment:
Possibly missing await here.

Reply:
it's a change from another branch, resolved in that branch